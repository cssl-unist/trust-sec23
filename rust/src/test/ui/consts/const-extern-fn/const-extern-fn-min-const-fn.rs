#![feature(const_extern_fn)]

const extern fn unsize(x: &[u8; 3]) -> &[u8] { x }
const unsafe extern "C" fn closure() -> fn() { || {} }
//~^ ERROR function pointer
//~| ERROR function pointer cast
const unsafe extern fn use_float() { 1.0 + 1.0; }
//~^ ERROR floating point arithmetic
const extern "C" fn ptr_cast(val: *const u8) { val as usize; }
//~^ ERROR casting pointers to integers


fn main() {}
