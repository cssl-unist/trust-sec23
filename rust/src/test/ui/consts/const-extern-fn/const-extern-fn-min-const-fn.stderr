error[E0658]: function pointers cannot appear in constant functions
  --> $DIR/const-extern-fn-min-const-fn.rs:4:41
   |
LL | const unsafe extern "C" fn closure() -> fn() { || {} }
   |                                         ^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn_fn_ptr_basics)]` to the crate attributes to enable

error[E0658]: function pointer casts are not allowed in constant functions
  --> $DIR/const-extern-fn-min-const-fn.rs:4:48
   |
LL | const unsafe extern "C" fn closure() -> fn() { || {} }
   |                                                ^^^^^
   |
   = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
   = help: add `#![feature(const_fn_fn_ptr_basics)]` to the crate attributes to enable

error[E0658]: floating point arithmetic is not allowed in constant functions
  --> $DIR/const-extern-fn-min-const-fn.rs:7:38
   |
LL | const unsafe extern fn use_float() { 1.0 + 1.0; }
   |                                      ^^^^^^^^^
   |
   = note: see issue #57241 <https://github.com/rust-lang/rust/issues/57241> for more information
   = help: add `#![feature(const_fn_floating_point_arithmetic)]` to the crate attributes to enable

error[E0658]: casting pointers to integers in constant functions is unstable
  --> $DIR/const-extern-fn-min-const-fn.rs:9:48
   |
LL | const extern "C" fn ptr_cast(val: *const u8) { val as usize; }
   |                                                ^^^^^^^^^^^^
   |
   = note: see issue #51910 <https://github.com/rust-lang/rust/issues/51910> for more information
   = help: add `#![feature(const_raw_ptr_to_usize_cast)]` to the crate attributes to enable

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0658`.
